### ISSUE TITLE
problem with external packages

### ISSUE THREAD

**Issue by jerrypgreenberg** [spack/spack#15771]
Hi, I am very new to spack

I was trying to get spack to use an already installed mpi installation for abinit.

I put into my yaml file:

packages:
  mvapich2_ib:
    version: [2.3.2]
    paths:
      mvapich2_ib@2.3.2%intel@18.0.1 arch=linux-centos7-haswell: /opt/mvapich2/intel/ib
    buildable: False
  all:
    providers:
      mpi: [ mvapich2_ib, openmpi_ib, intelmpi ]

and I edited the provided packages.py file for abinit to make the following change:

depends_on('mvapich2_ib', when='+mpi')

When I type:

spack install abinit

I get:

==> Error: Package 'mvapich2_ib' not found.

I don't know how to debug this.

Thanks

Jerry

**Comment by scheibelp** [comment]
Based on your example config, it looks like you have an instance of `mvapich2` that was compiled with Intel, in which case `mvapich2@2.3.2%intel@18.0.1 arch=linux-centos7-haswell` should work.

When you create an external reference, the package name in the spec must match a package that exists in Spack's package repository (for `mvapich2_ib@2.3.2%intel@18.0.1 arch=linux-centos7-haswell`, the package name is `mvapich2_ib`).

In other words, Spack must always have a package template for a package that is being installed (even if that package is not provided by Spack), the package name identifies the template.

**Comment by alalazo** [comment]
This was answered by Peter, and doesn't seem to be an issue. Closing.
